spec/unit/actions/v3/service_instance_create_managed_spec.rb

```diff
@@ -104,16 +104,112 @@ module V3
           end
         end
 
-        describe 'name already in use' do
-          before do
-            ManagedServiceInstance.make(name: name, space: space)
+        context 'when service instance with the same name already exists' do
+          let(:instance) { ManagedServiceInstance.make(name: name, space: space) }
+
+          context "when the last operation is in state 'create in progress'" do
+            before do
+              instance.save_with_new_operation({}, { type: 'create', state: 'in progress' })
+            end
+
+            it 'should raise' do
+              expect { action.precursor(message: message) }.to raise_error(
+                ServiceInstanceCreateManaged::InvalidManagedServiceInstance,
```

Comment:
IntelliJ has a strange sense of aesthetics. I would favor the following formatting:
```
              expect { action.precursor(message: message) }.to raise_error(
                ServiceInstanceCreateManaged::InvalidManagedServiceInstance,
                'The service instance name is taken: si-test-name.'
              )
```

Reply:
Indeed, fixed that.